src/TLuaInterpreter.cpp

```diff
@@ -11169,6 +11170,11 @@ int TLuaInterpreter::setIrcServer(lua_State* L)
     if (args > 2) {
         secure = getVerifiedBool(L, __func__, 3, "secure {default = false}", true);
     }
+    if (args > 3) {
+        if(!lua_isnoneornil(L,4)) {
+            password = getVerifiedString(L, __func__, 4, "server password {default = empty string}", true);
+        }
+    }
```

Comment:
I'd just collapse it to this, since if they provided a 4th argument, then they don't need to know that it is optional - they were intending in providing it already:

```suggestion
    if (args > 3) {
            password = getVerifiedString(L, __func__, 4, "server password", true);
    }
```

What do you think?

Reply:
I need to look closer when I'm fully awake.  I think the intent was to allow the user to use nil as one of the arguments and have it be treated as a valid way of saying no password, or to leave the password unchanged.  That, or I copied and pasted too much off of whatever I was looking at.